main.c

```diff
@@ -139,7 +160,12 @@ static void cmd_exec(char *param) {
 		swayidle_log_errno(LOG_ERROR, "fork failed");
 	} else {
 		swayidle_log(LOG_DEBUG, "Spawned process %s", param);
-		waitpid(pid, NULL, 0);
+		if (state.wait) {
+			swayidle_log(LOG_DEBUG, "Blocking until process exits");
+		}
+		int status;
+		waitpid(pid, &status, 0);
+		swayidle_log(LOG_DEBUG, "Process exit status: %d", WEXITSTATUS(status));
```

Comment:
It only makes sense to log this if `state.wait` is enabled. Otherwise the exit code will always be 0 (except if something went horribly wrong -- but we have other log messages to catch this). Also at this point the real child process wouldn't have exited.

Reply:
Good catch. I guarded the log message.